### PR TITLE
Multi-stage build, shrink production container by 60%

### PR DESCRIPTION
**Pull request by johnwalicki**
This Dockerfile uses a multi-stage build to shrink the production container from 291MB to 119MB.
It adopts registry.access.redhat.com/ubi8/nodejs-14-minimal:latest as the base image for production

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
 RUN mkdir -p /opt/app-root/src
 WORKDIR /opt/app-root/src
 COPY package.json /opt/app-root/src
-RUN npm install --only=prod
+RUN npm install --no-audit --no-update-notifier --no-fund --production
 COPY . .
 
-ENV NODE_ENV production
+## Release image
+FROM registry.access.redhat.com/ubi8/nodejs-14-minimal:latest
```
Comment (youming-lin) [inline Dockerfile:14]: Can you pin the image version here? I tested this file and got this error in dockerlint:
```
Dockerfile found at: ./Dockerfile
Linting Dockerfile
./Dockerfile:14 DL3007 Using latest is prone to errors if the image will ever update. Pin the version explicitly to a release tag
```

Comment (johnwalicki) [reply Dockerfile]: Yeah - hate that linter warning.

**Comment by youming-lin** [comment]
LGTM

**Review by youming-lin — APPROVED** [review]

**Comment by johnwalicki** [comment]
Not certain why this PR is trying to merge other git history.  Is that normal?

**Comment by youming-lin** [comment]
Could be from a merge from `development` to your personal branch?

**Comment by johnwalicki** [comment]
Final comment on this - The IBM container registry service provides 512MB of storage in the Free tier.  Keeping the node-red-app images small will save our users $$$.  It also prevents the deployment pipelines from failing unexpectedly (the error message is clear that your account exceeds your quota) but its still a failure.

**Pull request merged**